roundrobin/rr.go

```diff
@@ -62,14 +71,35 @@ func (r *RoundRobin) Next() http.Handler {
 }
 
 func (r *RoundRobin) ServeHTTP(w http.ResponseWriter, req *http.Request) {
-	url, err := r.NextServer()
-	if err != nil {
-		r.errHandler.ServeHTTP(w, req, err)
-		return
-	}
 	// make shallow copy of request before chaning anything to avoid side effects
 	newReq := *req
-	newReq.URL = url
+	stuck := false
```

Comment:
Hum, there is an issue here I think.
You don't call `url, err := r.NextServer()` if `stuck` is false.